src/transaction/log_recovery_redo.hpp

```diff
@@ -644,6 +645,13 @@ void log_rv_redo_record_sync_apply (THREAD_ENTRY *thread_p, log_rv_redo_context
 			 VPID_AS_ARGS (&rcv_vpid), (int) rcv.offset, (int) rcv.length);
     }
 
+  if (is_passive_transaction_server() && log_data.rcvindex == RVPGBUF_DEALLOC)
+    {
+      // rcv pgptr is already unfixed and invalidated by the deallocation (pgbuf_rv_dealloc_redo ())
+      assert (rcv.pgptr != nullptr && pgbuf_get_fix_count (rcv.pgptr) <= 0);
+      rcv.pgptr = nullptr;
+    }
+
   if (rcv.pgptr != nullptr)
     {
       pgbuf_set_lsa (thread_p, rcv.pgptr, &record_info.m_start_lsa);
```

Comment:
This is out of the scope of this issue, but I think we should advance the page lsa before unfixing the page to redo it correctly. If it's the case, it needs an issue for it.

Reply:
I was looking for reasons to set the page LSA in PTS. However, not only are the reasons important, but it also seems odd that when the same page has the same LSA applied, ATS, PS, and PTS have different Page LSAs. I think it's inappropriate to set the page LSA for the same page_ptr after invalidating the page. According to your opinion, I think it's necessary to leave comments and create an issue for resolution.

@cristiarg  What is your opinion?

Reply:
To be honest, I'm not quite sure about other reasons. It doesn't address the deallocate-reallocate situation, and after the corresponding page buffer is invalidated, when PTS read the same page, it obtains page from PS, so it seems the page LSA won't be used in PTS. However, I aimed to eliminate potential errors by making it behave similarly to ATS and PS.

Reply:
>  it obtains page from PS, so it seems the page LSA won't be used in PTS

When determining the LSA for fetching a page (`pgbuffer_pool.highest_evicted_lsa`), the last evicted page's stuff is used. The PTS expects the requested PS to apply redo log records up to that point. Let's imagine the last evicted page is the deallocated page by the replicator in `pgvbuf_rv_dealloc_redo`. It makes a change to the page, but the newly fetched page doesn't have the modification since the corresponding page_lsa is not updated. 

Can't it happen?

Reply:
> When determining the LSA for fetching a page (pgbuffer_pool.highest_evicted_lsa), the last evicted page's stuff is used

As far as I know, PTS uses `highest_process_lsa` instead of `highest_evicted_lsa`. 
Therefore, when the PTS requests a page, the PS is expected to return a page that includes the changes up to `RVPGBUF_DEALLOC`

code snippet : 
```
in pgbuf_read_page_from_file_or_page_server ()
...
 8206       const LOG_LSA target_repl_lsa = is_passive_transaction_server ()?
 8207         get_passive_tran_server_ptr ()->get_highest_processed_lsa () : pgbuf_Pool.get_highest_evicted_lsa ();
...
```

Reply:
Oh, you're correct. I was confused.